### PR TITLE
[feature] Added browser based tests using selenium #237

### LINKED ISSUE #237: [tests] Add browser based test (eg: Selenium)
Related #213.

Implement basic browser testing with selenium for the following features:

- signup success
- signup failure (validation error)
- login success
- login failure
- status

### PR DESCRIPTION
**Pull request by codesankalp**
Added password-change.test.js

Closes #237

### CODE REVIEW

**Review by nemesisdesign — COMMENTED** [review]
I think then just adding mobile verification flow (with one failed attempt on purpose and then 
 a successful attempt) and change mobile phone number will be enough :+1:.

**Comment by codesankalp** [comment]
> I think then just adding mobile verification flow (with one failed attempt on purpose and then
> a successful attempt) and change mobile phone number will be enough.

I am working on it, but it will fail in CI because the mobile verification configuration of the new organization is needed before running `yarn start`.

File: README.md [hunk 1]
```diff
@@ -130,7 +130,12 @@ Prerequisites for running browser-based tests:
...
 4. If a virtual environment is used to run openwisp-radius then
    this needs to be activated before running browser tests.
-5. In the test environment of openwisp-radius, the `default` organization
+5. Configuration file of `mobile` organization is needed before running `yarn start`.
+   `mobile` organization can be created by running:
+   ```
+   yarn add-org --noprompt '{"name":"mobile","slug":"mobile","uuid":"organization_uuid","secret_key":"organization_secret_key","mobile_phone_verification":true,"subscriptions":false,"login_action_url":"http://localhost:8000/captive-portal-mock/login/","logout_action_url":"http://localhost:8000/captive-portal-mock/logout/","logout_by_session_ID":true,"remember_me":true,"openwisp_radius_url":"http://127.0.0.1:8000","assets_confirm":true}'
+   ```
+6. In the test environment of openwisp-radius, the `default` organization
    must be present or alternatively
```
Comment (nemesisdesign) [inline README.md:139]: BTW I noticed an unfinished sentence here, can you please remove the unfinished part?

**Review by nemesisdesign — COMMENTED** [review]
Looks good, it runs, I tried it also locally, I will need a bit more time to review the code to double check that all new tests are good.

Please see my comment below.

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Can you rebase on the latest master please?

**Review by nemesisdesign — COMMENTED** [review]
@codesankalp is there a chance you can an useful test for the language switcher?

**Comment by codesankalp** [comment]
> @codesankalp is there a chance you can an useful test for the language switcher?

Can we change the default configuration file (to add Italiano as a language in the header)?

**Comment by nemesisdesign** [comment]
> > @codesankalp is there a chance you can an useful test for the language switcher?
> 
> Can we change the default configuration file (to add Italiano as a language in the header)?

@codesankalp if we can we do it only in the browser tests for sure

File: browser-test/password-confirm.test.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    driver = await getDriver();
+  }, 30000);
+
+  afterAll(async () => {
+    await clearData();
+    await driver.manage().deleteAllCookies();
+    driver.close();
+  });
+
+  it("should show not found on password confirm", async () => {
```
[the remark this answers is not in the document]
Comment (codesankalp) [reply browser-test/password-confirm.test.js:25]: It tests the failure of password-confirm when uid and token are invalid.

File: browser-test/mobile-verfication.test.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    const token = getPhoneToken();
+    codeInput.clear();
+    codeInput.sendKeys(token);
+    submitBtn = await getElementByCss(driver, "button[type='submit']");
+    await driver.wait(until.elementIsVisible(submitBtn));
+    submitBtn.click();
+    await getElementByCss(driver, "div#status");
+    const emailElement = await getElementByCss(
+      driver,
+      "div > p:nth-child(5) > span",
```
[the remark this answers is not in the document]
Comment (codesankalp) [reply browser-test/mobile-verfication.test.js:76]: I tried but this is the one that is working.

**Review by nemesisdesign — APPROVED** [review]

**Pull request merged**